contrib/google-cloud-cpp-cmake/cmake/CompileProtos.cmake

```diff
@@ -0,0 +1,527 @@
+        "google-cloud-cpp::identity_accesscontextmanager_type_type_protos")
+    # Replace "google-cloud-cpp::$1" with "google-cloud-cpp:$2" in deps. The
+    # most common reason to need one of these is a dependency between the protos
+    # in one library vs. the protos in a second library. The AIPs frown upon
+    # such dependencies, but they do happen.
+    set(target_substitutions
+        "grafeas_v1_grafeas_protos\;grafeas_protos"
+        "iam_v2_policy_protos\;iam_v2_protos"
+        "logging_type_type_protos\;logging_type_protos"
+        "identity_accesscontextmanager_v1_accesscontextmanager_protos\;accesscontextmanager_protos"
+        "cloud_osconfig_v1_osconfig_protos\;osconfig_protos"
+        "cloud_documentai_v1_documentai_protos\;documentai_protos")
+
+    foreach (line IN LISTS contents)
+        if ("${line}" STREQUAL "")
+            continue()
+        endif ()
+        string(REPLACE ":" "_" line "${line}")
+        string(REPLACE "_proto" "_protos" line "${line}")
+        string(REPLACE "@com_google_googleapis//" "google-cloud-cpp::" line
+                       "${line}")
+        # Avoid duplicate `google`'s in the target name.
+        string(REPLACE "google-cloud-cpp::google/" "google-cloud-cpp::" line
+                       "${line}")
+        string(REPLACE "/" "_" line "${line}")
+        if ("${line}" IN_LIST targets_to_omit)
+            continue()
+        endif ()
+        foreach (substitution IN LISTS target_substitutions)
+            set(from_to "${substitution}")
+            list(GET from_to 0 from)
+            list(GET from_to 1 to)
+            string(REPLACE "google-cloud-cpp::${from}"
+                           "google-cloud-cpp::${to}" line "${line}")
+        endforeach ()
+        list(APPEND deps "${line}")
+    endforeach ()
+    set(${var}
+        "${deps}"
+        PARENT_SCOPE)
+endfunction ()
+
+# include(GNUInstallDirs)
+
+# Install headers for a C++ proto library.
+function (google_cloud_cpp_install_proto_library_headers target)
+    cmake_parse_arguments(_opt "" "OUT_DIRECTORY" "" ${ARGN})
+    google_cloud_cpp_set_out_directory(_opt_OUT_DIRECTORY OUT_DIR)
+    get_target_property(type ${target} TYPE)
+    if ("${type}" STREQUAL "INTERFACE_LIBRARY")
+        return()
+    endif ()
+    get_target_property(target_sources ${target} SOURCES)
+    foreach (header ${target_sources})
+        # Skip anything that is not a header file.
+        if (NOT "${header}" MATCHES "\\.h$")
+            continue()
+        endif ()
+        string(REPLACE "${OUT_DIR}/" "" relative "${header}")
+        get_filename_component(dir "${relative}" DIRECTORY)
+        install(
+            FILES "${header}"
+            DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${dir}"
+            COMPONENT google_cloud_cpp_development)
+    endforeach ()
+endfunction ()
+
+# Install protos for a C++ proto library.
+function (google_cloud_cpp_install_proto_library_protos target source_dir)
```

Comment:
This function seems similar to `google_cloud_cpp_add_common_options`. I think we shouldn't use this.